### PR TITLE
fix(sdk/python): restore telemetry initialization in modules

### PR DESCRIPTION
**Pull request by helderco**
Fixes https://github.com/dagger/dagger/issues/7673

### CODE REVIEW

File: sdk/python/src/dagger/client/_session.py [hunk 1]
```diff
@@ -172,6 +172,7 @@ def __await__(self):
         return self.connect().__await__()
 
     async def __aenter__(self) -> Self:
+        telemetry.initialize()
```
Comment (helderco) [inline sdk/python/src/dagger/client/_session.py:175]: The initialization was setup initially while the SDK establishes a connection to the API, but it was added only to the connection that has automatic provisioning support. So added to this one too, which is the one used in modules.

File: sdk/python/src/dagger/mod/_module.py [hunk 2]
```diff
@@ -189,39 +189,31 @@ def __call__(self) -> None:
...
+            msg = f"Failed to serialize result: {e}"
+            raise InternalError(msg) from e
+
+        logger.debug(
+            "output => %s",
+            textwrap.shorten(repr(output), 144),
+        )
+        await dag.current_function_call().return_value(dagger.JSON(output))
 
-    @serve()
```
Comment (helderco) [inline sdk/python/src/dagger/mod/_module.py:224]: The decorator was created because of the spans, so removing again.

**Review by helderco — COMMENTED** [review]
The SDK used to create an OTel span to replace `exec /runtime`. It was removed in  https://github.com/dagger/dagger/pull/7452, which inadvertently also disconnected the telemetry initialization.

**Comment by skycaptain** [comment]
I can confirm that the changes fix issue #7673.

**Review by vito — APPROVED** [review]
Whoops, thanks for fixing! :pray:

**Pull request merged**